screens/supporter-skillimpact-screen/SupporterSkillImpactScreen.js

```diff
@@ -1,38 +1,109 @@
 import React, { Component } from 'react';
-import { View } from 'react-native';
+import { View, StyleSheet, Alert } from 'react-native';
+import { ScrollView } from 'react-navigation';
+import * as SecureStore from 'expo-secure-store';
+import {
+  getCampaignsBySkill,
+  getProfileData,
+  getApplicationsByUser,
+} from '../../store/actions';
+import { connect } from 'react-redux';
+import SupporterSkilledImpactHeader from '../../components/SkilledImpact/SupporterSkilledImpactHeader';
+import SupporterSkilledImpactBody from '../../components/SkilledImpact/SupporterSkilledImpactBody';
 
 class SupporterSkillImpactScreen extends Component {
   static navigationOptions = ({ navigation }) => {
-    //TODO use navigations to establish stack navigation logic
     return {
-      title: 'Supporter Skill Impact',
+      title: 'My Skilled Impact',
       headerStyle: {
-        backgroundColor: '#323338'
+        backgroundColor: '#323338',
       },
       headerTintColor: '#fff',
       headerLeft: () => <View />,
     };
   };
   constructor(props) {
-    //TODO props edits and states
     super(props);
+    this.state = {
+      campaigns: [],
+      submissions: [],
+      skills: [],
+      userId: 0,
+      isAcceptingHelp: false,
+      loading: true,
+    };
+    this.initProfileData = this.initProfileData.bind(this);
   }
 
   componentDidMount() {
-    //TODO when the screen is loaded
+    this.initProfileData();
   }
 
+  initProfileData = async () => {
+    try {
+      const userId = await SecureStore.getItemAsync('id', {});
+      await this.props.getProfileData(userId, null, true);
+      let { skills, accepting_help_requests } = this.props.currentUserProfile;
+      // Placeholder for skills, remove once Skills Creation screen is made
+      if (!skills) {
+        skills = ['ARCHITECTURE', 'DRONE'];
+      }
+      skills.forEach(async (skill) => {
+        await this.props.getCampaignsBySkill(skill);
+        this.setState({
+          campaigns: this.state.campaigns.concat(this.props.campaignsBySkill),
+        });
+      });
```

Comment:
You can edit your skills in the edit profile page (right-most tab, gear icon in top right) now, so can we get rid of this?